File: tests/week01/ReverseString.java
package tests.week01;

public class ReverseString {
    
public static void main(String[] args) {

String str = "a";
String reversed = "";

/*if (str == null){
    System.out.println("String cannot be null");
    return;

} */

if (str.isEmpty()){
    System.out.println("String cannot be empty");
    return ;
}

if(str.length() == 1){
    System.out.println("String needs to be at least two characters to reverse it");
    return;
}

for (int i = str.length()-1; i >= 0 ; i--) {
    reversed += str.charAt(i);
}

System.out.println(reversed);

}



}

File: tests/week01/reverseString.js



function reverseString(str){

let reversed = "";

if(str == null || str == undefined){
    console.log("String can not be null or undifined");
    return;
}

if(str.length == 0 ){
    console.log("String cannot be empty");
    return;
}

for (let i = str.length()-1; i >= 0; i--){
    reverseString += str(i);
}
console.log(reverseString);
}